src/java.base/share/classes/java/security/CodeSource.java

```diff
@@ -456,12 +457,16 @@ private boolean matchLocation(CodeSource that) {
                     return false;
                 }
                 if (this.sp == null) {
-                    this.sp = new SocketPermission(thisHost, "resolve");
+                    @SuppressWarnings("removal")
+                    var _ = this.sp = new SocketPermission(thisHost, "resolve");
                 }
                 if (that.sp == null) {
-                    that.sp = new SocketPermission(thatHost, "resolve");
+                    @SuppressWarnings("removal")
+                    var _ = that.sp = new SocketPermission(thatHost, "resolve");
                 }
-                return this.sp.implies(that.sp);
+                @SuppressWarnings("removal")
+                var result = this.sp.implies(that.sp);
```

Comment:
Will we need to rewrite this code without SocketPermissions? Can we remove it, maybe? It is only used by CodeSource.implies, which is not used by the JDK (but may be used elsewhere).

Reply:
also the JavaDoc of the implies method references the SocketPermission. That will also need to be cleaned up at some point.

Reply:
That will be left over for someone in security-libs to cleanup this code at their convenience before we remove SocketException.

Hmmm... If the public API references SocketException we might have to deprecate in this PR too. Not sure what the implications are. Maybe @seanjmullan can advise.